grpc/dispatcher.go

```diff
@@ -0,0 +1,172 @@
+				Code:    int(responseStatus.Code()),
+				Message: responseStatus.String(),
+			})
+	}
+
+	responseByte, err := proto.Marshal(responseProto)
+	if err != nil {
+		return fiber.NewErrorResponse(err)
+	}
+
+	return &Response{
+		Metadata:      responseHeader,
+		Status:        *status.New(codes.OK, "Success"),
+		CachedPayload: fiber.NewCachedPayload(responseByte),
+	}
+}
+
+// NewDispatcher is the constructor to create a dispatcher. It will create the clientconn and set defaults.
+// Endpoint, serviceMethod and response proto are required minimally to work.
+func NewDispatcher(config DispatcherConfig) (*Dispatcher, error) {
+	configuredTimeout := TimeoutDefault
+	if config.Timeout != 0 {
+		configuredTimeout = config.Timeout
+	}
+
+	if config.Endpoint == "" || config.Service == "" || config.Method == "" || config.ResponseProtoName == "" {
+		return nil, fiberError.ErrInvalidInput(
+			protocol.GRPC,
+			errors.New("grpc dispatcher: missing config (endpoint/service/method/response-proto)"))
+	}
+
+	conn, err := grpc.DialContext(context.Background(), config.Endpoint, grpc.WithTransportCredentials(insecure.NewCredentials()))
+	if err != nil {
+		// if ok is false, unknown codes.Unknown and Status msg is returned in Status
+		responseStatus, _ := status.FromError(err)
+		return nil, fiberError.ErrRequestFailed(
+			protocol.GRPC,
+			errors.New("grpc dispatcher: "+responseStatus.String()))
+	}
+
+	// create a reflection client and get FileDescriptorProtos
+	reflectionClient := grpc_reflection_v1alpha.NewServerReflectionClient(conn)
+	req := &grpc_reflection_v1alpha.ServerReflectionRequest{
+		MessageRequest: &grpc_reflection_v1alpha.ServerReflectionRequest_FileContainingSymbol{
+			FileContainingSymbol: config.Service,
+		},
+	}
+	reflectionInfoClient, err := reflectionClient.ServerReflectionInfo(context.Background())
+	if err != nil {
+		return nil, fiberError.NewFiberError(
+			protocol.GRPC,
+			errors.New("grpc dispatcher: unable to get reflection information, ensure server reflection is enable and config are correct"))
+	}
+	if err = reflectionInfoClient.Send(req); err != nil {
+		return nil, fiberError.NewFiberError(protocol.GRPC, err)
+	}
+	reflectionResponse, err := reflectionInfoClient.Recv()
+	if err != nil {
+		return nil, fiberError.NewFiberError(protocol.GRPC, err)
+	}
+
+	// From the FileDescriptorProtos, find the proto matching config proto name
+	var fileDescriptorProtos []*descriptorpb.FileDescriptorProto
+	for _, fdBytes := range reflectionResponse.GetFileDescriptorResponse().FileDescriptorProto {
+
+		fd := &descriptorpb.FileDescriptorProto{}
+		if err := proto.Unmarshal(fdBytes, fd); err != nil {
+			return nil, fiberError.NewFiberError(protocol.GRPC, err)
+		}
+		fileDescriptorProtos = append(fileDescriptorProtos, fd)
+	}
+	var fdProto *descriptorpb.FileDescriptorProto
+	for _, fd := range fileDescriptorProtos {
+		for _, md := range fd.MessageType {
+			if strings.Contains(*md.Name, config.ResponseProtoName) {
```

Comment:
Related to https://github.com/gojek/fiber/pull/3/files#r958023489. I think we should also check the package name properly, in case there are 2 protos with same name but located in different package.